internal/exec/stages/files/passwd.go

```diff
@@ -100,9 +100,12 @@ func (s stage) createUsers(config types.Config) error {
 	defer s.Logger.PopPrefix()
 
 	for _, u := range config.Passwd.Users {
-		if err := s.EnsureUser(u); err != nil {
+		shouldExist := u.ShouldExist == nil || *u.ShouldExist
```

Comment:
`EnsureUsers` already takes a `types.PasswdUser`. I think it would be cleaner to do the check `EnsureUser` and `CreateGroup` since all the create logic in the same function.

Reply:
Hmm, having the quick exit hatch of checking `shouldExist` after `EnsureUser` rather than baking in detection logic to `SetPasswordHash` & `AuthorizeSSHKeys` is valuable though so we'd probably want to know the status here.

Are you suggesting not passing it to `EnsureUser` and modifying the else to `else if !(u.ShouldExist == nil || *u.ShouldExist) && err == nil`?

Reply:
> Are you suggesting not passing it to EnsureUser and modifying the else to `else if !(u.ShouldExist == nil || *u.ShouldExist) && err == nil?`

More or less. It just seems strange to check `u.ShouldExists` while passing the `type.PasswdUser` completely to `s.EnsureUser`.